srcup/cli.py

```diff
@@ -54,6 +55,10 @@ async def asingle(artifact: CryticCompile, api_url: str, api_key: str, name: str
         bytecode_hashes = b"".join([item.codehash for item in bytecodes])
         git_hash = sha1(bytecode_hashes).hexdigest()
 
+    if name is None or not name:
+        path = pathlib.PurePath(target)
```

Comment:
I think this will have a problem with relative paths. `pathlib.Path` has a `resolve()` method that can be used to turn the underlying path to absolute (which should be ok to get the name from)

Reply:
changed